job/validators.py

```diff
@@ -209,42 +209,43 @@ def cross_clean(self):
         PV tilt set to latitude if not provided and prod_factor_series validated
         """
         def cross_clean_pv(pvmodel):
-            if pvmodel.__getattribute__("tilt") == 0.537:  # 0.537 is a dummy number, default tilt
-                pvmodel.__setattr__("tilt", self.models["Site"].__getattribute__("latitude"))
+            if pvmodel.__getattribute__("tilt") == None:  # 0.537 is a dummy number, default tilt # TODO: change to None?
+                if pvmodel.__getattribute__("array_type") == "ROOFTOP_FIXED":
+                    pvmodel.__setattr__("tilt", 10)
+                else:
+                    pvmodel.__setattr__("tilt", abs(self.models["Site"].__getattribute__("latitude"))) # TODO differentiate default tilt based on array_type
+            
+            if pvmodel.__getattribute__("azimuth") == None:
+                if self.models["Site"].__getattribute__("latitude") >= 0:
+                    pvmodel.__setattr__("azimuth", 180)
+                else:
+                    pvmodel.__setattr__("azimuth", 0.0)
+            
+            # TODO add cross_clean for azimuth
             if pvmodel.__getattribute__("max_kw") > 0:
                 if len(pvmodel.__getattribute__("prod_factor_series")) > 0:
                     self.clean_time_series("PV", "prod_factor_series")
+            
+            if self.models["Settings"].off_grid_flag==True:
+                self.models["PV"].can_net_meter = False
+                self.models["PV"].can_wholesale = False
+                self.models["PV"].can_export_beyond_nem_limit = False
+                self.models["PV"].operating_reserve_required_pct = 0.25
+            else:
+                if self.models["PV"].__getattribute__("can_net_meter") == None:
```

Comment:
instead of these 3 if statements, why not just make the defaults in models.py True, then only keep the above if off grid block not the else block?

Reply:
the help and inputs endpoints use those defaults in models.py so they won't show up in the web tool input boxes otherwise. Off grid isn't in the web tool right? otherwise we would need on and off grid versions of help.

Reply:
@hdunham off-grid is in the web tool. You uncheck "Grid" to model off-grid.

Reply:
I have updated the PV field validations (thank you Hallie!) for `can_net_meter`, `can_export_beyond_nem_limit`, `can_wholesale`, so we default these to True in models.py but set them to False if off-grid is true.

Currently, `operating reserve required pct` is not defaulting to any value. We set this field to 0.25 if off-grid is true, and if off-grid is false, we override anything the user might have provided and set this value to 0.0 per [here](https://github.com/NREL/REopt.jl/blob/8d7beb8a916d1cdf8571c268d44a6ef86dc07a7b/src/core/pv.jl#L176) (defaulting it will allow user to provide an input value, which we dont want). I have also updated the help text for this field so users know this gets over-written regardless of what they provide (if they provide a value).

@adfarth we can know if user provided an input by checking if `operating_reserve_required_pct` is None. But it is my understanding that we dont want any ability to modify this field? It has to be 0.25 for off-grid and 0.0 for on grid? I am not seeing us enforcing this field to be 25% in julia though.

Reply:
@rathod-b see my comments below regarding `operating_reserve_required_pct` for `PV, Wind, and Load`. These should **not** get overwritten when `off_grid_flag=True`. They should default to their respective values (shown in Julia code) but the user can modify these inputs for off-grid scenarios.

Reply:
Implemented in https://github.com/NREL/REopt_API/pull/343/commits/152ac98842eaaf16a7068f0412b34b0893f0886a

Reply:
Yep I mostly agree with the situations you laid out @adfarth . Sometimes we need to know if the user didn't enter a value, but for the inputs where in one grid case we take the user input/default and the other grid case we override no matter what then we don't need to know. 
The one part I don't completely agree on is the second situation you listed, which applies to operating_reserve_required_pct. Technically, we don't need to know if the user provided a value. We could make the Django default equal to off-grid default and if offgrid=false then override it to 0. Essentially just the reverse of the first situation in your list. We might not want to do this because then there would be a mix of on- and off- grid defaults in the inputs endpoint. Or, maybe that's fine because the help text explains when an input is only applicable to off-grid. Just wanted to point out that there are two options to decide between, each with pros and cons.

Reply:
I also left a couple comments on help text in [https://github.com/NREL/REopt_API/commit/152ac98842eaaf16a7068f0412b34b0893f0886a](url)

Reply:
I also left a couple comments on help text in [152ac98](https://github.com/NREL/REopt_API/commit/152ac98842eaaf16a7068f0412b34b0893f0886a)

Reply:
When I said having on and off grid versions of help, I mean of the help/ endpoint (equivalent to inputs/ endpoint). That would mean creating a new function in views.py.

Reply:
@hdunham I agree with the logic you've laid out! I have been thinking that we wouldn't want to mix on- and off-grid defaults in `models.py`, though. I think we should ask the broader team about this

Reply:
I made an issue so we can remember to address this later as it's lower priority at the moment I think.